scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -470,8 +470,10 @@ class Router(formatOps: FormatOps) {
           Split(NewlineT(isDouble = tok.hasBlankLine), 0)
         )
 
-      case tok @ FormatToken(_: T.RightParen, _: T.KwDef | _: T.Comment, _)
-          if leftOwner.is[Defn.ExtensionGroup] =>
+      case tok @ FormatToken(_: T.RightParen, _, _)
```

Comment:
this change, and its tests, seem unrelated to infix, no? should they be part of a different pr?

Reply:
At least `.isNot[LeftParenOrBrace]` is needed for this change and since the list of accepted tokens started growing it turned out to be easier to just use it together with `nextNonComment`, which simplifies this logic a lot.

Because, this also fixes the issue with the comments I added the tests here. I can add it in a different PR, but it would just be a PR with those tests.

Reply:
yes, i understand now. what type do you get for rightOwner of next non comment when it is def, infix, (, {, etc?

Reply:
it should be `Defn.ExtensionGroup`  for `(`, `Term.Block` for `{` for the others `Defn.Def` - so nothing I think to help us here.